app/controllers/topics_controller.rb

```diff
@@ -0,0 +1,72 @@
+class TopicsController < ApplicationController
+  before_action :set_topic, only: %i[ show edit update destroy ]
+
+  # GET /topics or /topics.json
+  def index
+    @topics = Topic.all
+    render json: @topics
+  end
+
+  # GET /topics/1 or /topics/1.json
+  def show
+    render json: @topics
+  end
+
+  # GET /topics/new
+  def new
+    @topic = Topic.new
+  end
+
+  # GET /topics/1/edit
+  def edit
+  end
+
+  # POST /topics or /topics.json
+  def create
+    @topic = Topic.new(topic_params)
+
+    respond_to do |format|
+      if @topic.save
+        format.html { redirect_to topic_url(@topic), notice: "Topic was successfully created." }
+        format.json { render :show, status: :created, location: @topic }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @topic.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /topics/1 or /topics/1.json
+  def update
+    respond_to do |format|
+      if @topic.update(topic_params)
+        format.html { redirect_to topic_url(@topic), notice: "Topic was successfully updated." }
+        format.json { render :show, status: :ok, location: @topic }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @topic.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /topics/1 or /topics/1.json
+  def destroy
+    @topic.destroy
+
+    respond_to do |format|
+      format.html { redirect_to topics_url, notice: "Topic was successfully destroyed." }
+      format.json { head :no_content }
+    end
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_topic
+      @topic = Topic.find(params[:id])
+    end
```

Comment:
You can memoize the variable, see https://www.honeybadger.io/blog/ruby-rails-memoization/

```suggestion
    def topic
      @topic ||= Topic.find(params[:id])
    end
```

Reply:
so the idea here is that the topic will not change often so it makes sense to cache it?

Reply:
I think the idea is: not calculate the topic again, the Topic.find() is executed only one time.